blips/blip-0002.mediawiki

```diff
@@ -0,0 +1,106 @@
+<pre>
+bLIP: 2
+Title: Keysend
+Status: Active
+Type: Standards Track
+Author: Valentine Wallace <vwallace@protonmail.com>
+Created: 2021-05-21
+License: BSD-2-Clause, OPL
+</pre>
+
+==Abstract==
+
+Keysend is a type of lightning payment that does not require the payee to
+provide an invoice. Instead, the payer includes their payer-selected payment
+preimage in the TLV onion.
+
+As of authoring this bLIP, keysend is already supported by lnd, C-Lightning,
+Eclair, LDK, and possibly others. This bLIP serves to document what is already
+well-supported in the wild, for posterity and so that new implementations don't
+have to reverse-engineer keysend from existing implementations.
+
+==Copyright==
+
+This bLIP is dual-licensed under the Open Publication License and BSD 2-clause
+license.
+
+==Specification==
+
+Sender:
+* MUST include a TLV record keyed by type `5482373484` with a TLV value of a
+  randomly generated and cryptographically-secure 32-byte value that serves as
+  the HTLC payment preimage
+* MUST NOT set a `payment_data` field
```

Comment:
What's `payment_data` here?

Reply:
From [this section](https://github.com/lightningnetwork/lightning-rfc/blob/master/04-onion-routing.md#tlv_payload-format). I clarified a bit.